src/v/redpanda/application.cc

```diff
@@ -421,6 +421,13 @@ void application::initialize(
 
     _abort_on_oom->watch(oom_config_watch);
 
+    construct_service(
+      _cpu_profiler,
+      config::shard_local_cfg().cpu_profiler_enabled.bind(),
```

Comment:
I think you need to wrap this in a `ss::sharded_paramter` otherwise you will only dynamically enable/disable on shard 0.

Reply:
D'oh, right you are. Switching to `ss::sharded_parameter`.

Reply:
Isn't it actually that the "bound property" object for shard 0 will be passed to all shards, which I guess may "work" but is a thread-safety violation?

Reply:
Yeah, it was working, but was definitely a thread-safety violation. I suppose other shards could see a partial write potentially. Though I don't know any arch that writes bools non-atomically.